### DISCUSSION TITLE
How can I ensure consistency within tags?

### DISCUSSION THREAD

**Discussion by DanMcAdam** [inkle/ink#835] [Game integration help] [answered]
Is there a way to normalize tags? It seems like a necessity for their purpose, since writers will be writing potentially hundreds or thousands of lines of dialogue, many of which have an accompanying in game animation, noise, etc. and string comparison is notoriously finicky, a misremembered animation name ("#animation:pensive" instead of "animation:reserved" for example) or a simple capitalization error won't show up as an error until you try to run the code. Storing tags as variables, or placing variables in tags, would solve this problem. Is there a way to do something like that?

**Comment by joningold** [comment]
You can use variables and constants in tags.

CONST ANIM=“animation”
LIST animations = pensive, excited

JIM: let’s see now… #{ANIM}: {pensive}

**Reply by DanMcAdam** [reply]
The editor's autocomplete doesn't work after a tag for some reason, is there a way to fix that? It does at least provides error checking when the variable is wrong, but suggestions would be make it much more useful <img width="341" alt="Inky_rNab7TF1kN" src="https://user-images.githubusercontent.com/41479497/230707404-fa7e91d5-510a-4cfd-a01d-277ef44a86f7.png">
<img width="340" alt="Inky_JoL23vwGOd" src="https://user-images.githubusercontent.com/41479497/230707478-51009cc3-54cd-4e98-95ff-2cfd330973ff.png">

**Reply by DanMcAdam** [reply]
Actually, tags seem to go out as straight strings and aren't converted to variables. After experimenting with this, my parsing system is receiving `{TAG_SPEAKER}` when `CONST TAG_SPEAKER = "speaker:"`, so unless I'm misusing the tag system this won't work at all.

**Comment by joningold** [comment, marked as answer]
Adaptive tags are new as of ink 1.1 so you need to make sure you have latest. If you’re using the js export I think that’s still needing to be updated but there’s an informal patch available - you can ask on the discord for it.

**Comment by matthew-dean** [comment]
You can write tags with a list like:
```
LIST expression = neutral, smirking, surprised, angry, amused, playful, serious, impressed, thoughtful, sad, excited, skeptical, affectionate, stern, worried

#{expression.smirking}
```

This will guarantee that you don't mis-spell expression names by referencing a list.
